lnst/Devices/Device.py

```diff
@@ -41,6 +42,36 @@ def __instancecheck__(self, other):
         except AttributeError:
             return super(DeviceMeta, self).__instancecheck__(other)
 
+def sriov_capable(func):
+    """
+    Decorator for Device class methods that checks whether the Device is
+    capable of SR-IOV.
+    """
+    def wrapper(self, *args, **kwargs):
+        try:
+            exec_cmd(f"cat /sys/class/net/{self.name}/device/sriov_numvfs")
```

Comment:
> But on the other hand I understand that you basically only work with /sys/class/net... so it is probably fine

I'm not aware of other way how to configure vfs on a sriov capable device, so checking this file is not based on assumption, but simply a pre-requisite. If this file does not exist, the device can't create vfs. Sure, it may also mean that the device does not have the feature enabled - but in that case, it simply is not capable of using the sriov features, iow, is not sriov capable from the current device status point of view.

>  It could possibly make more sense to check lspci or similar bus info to see whether SRIOV is supported.

Not quite sure how I could fetch this info. Also, what options do we have if the device can have the SRIOV feature turned on but is disabled. We would have to enable such feature either in BIOS or using a driver specific tools which is imo out of scope.

Maybe I'm missing something, please clarify if that's the case.

Reply:
I can't find my comments from issues where I was verifying this, but I remember using `lspci -vv` on the given bus info to see whether it had "single root" capabilities. This ensured that BIOS settings for the card were enabled and that the presense of the `sriov_numvfs`.

However just the last assumption and existence should be enough. Disregard this.